SameDayReid/gallery_creation.py

```diff
@@ -70,46 +83,95 @@ def add_video_to_gallery(self, video_path:str, face_clf, skip_every=500):
-            crop_candidates_faces = []
-            for i, (id, conf, crop_im) in enumerate(zip(ids, confs, crops_imgs)):
-                try:
-                    ## first detection run --- detect all faces in image
-                    face_imgs, face_bboxes, face_probs = self.arc.detect_face_from_img(crop_img=crop_im)
-                    # if only a single face exists, apply a high threshold to get a high resolution face image and body
-                    if face_imgs is not None and len(face_imgs) == 1:
-                        face_img, face_prob = self.pose_estimator.find_matching_face(crop_im, face_bboxes, face_probs,
-                                                                                    face_imgs)
-                        if is_img(face_img) and face_prob >= 0.8: # hard coded min threshold for face rec
-                            crop_candidates_faces.append(face_img)
-                            crop_candidates_inds.append(i)
-                        # else: TODO write when pose estimation discards image
-
-                except Exception as e:
-                    # print(e)
-                    # print('err')
-                    continue
-            crop_cands = [crop_im for i, crop_im in enumerate(crops_imgs) if i in crop_candidates_inds]
-            if len(crop_candidates_faces) > 0: # some faces where detected
-                for i, (face,crop_im) in enumerate(zip(crop_candidates_faces, crop_cands)):
-                    # numpy_img = face.permute(1, 2, 0).int().numpy().astype(np.uint8)
-                    face = face[:, :, ::-1]
+            video_crops.extend(self.detect_faces_and_create_db_crops(ids=ids,
+                                                        confs=confs,
+                                                        crops_imgs=crops_imgs,
+                                                        crops_bboxes=crops_bboxes,
+                                                        vid_name=vid_name,
+                                                        part=part,
+                                                        frame_num=image_index
+                                                        ))
+
+        add_entries(crops=video_crops, db_location=SAME_DAY_DB_LOCATION) # Note this points to the SAME_DB_DB_LOCATION !
+
+    def detect_faces_and_create_db_crops(self, ids, confs, crops_imgs, crops_bboxes, vid_name, part, frame_num):
+        crops = []
+        pose_discard_counter = 0
+        errs_counter = 0
+        for i, (id, conf, crop_im, crop_bbox) in enumerate(zip(ids, confs, crops_imgs, crops_bboxes)):
+            # try:
+            ## first detection run --- detect all faces in image
+            face_imgs, face_bboxes, face_probs = self.arc.detect_face_from_img(crop_img=crop_im)
+            # if only a single face exists, apply a high threshold to get a high resolution face image and body
+            if face_imgs is not None and len(face_imgs) == 1:
```

Comment:
So currently we ignore crops in which more than 1 face was detected, right?

Reply:
Yep, in the gallery creation